File: 03-RecViz/recviz/backend/app/models/dataset.py
from __future__ import annotations

from app.models.base import CamelModel


class ColumnInfo(CamelModel):
    name: str
    type: str
    is_dimension: bool = True
    is_metric: bool = False
    filterable: bool = True


class DatasetInfo(CamelModel):
    id: int
    name: str
    table_name: str
    database_id: int
    columns: list[ColumnInfo] = []
    row_count: int | None = None


class SchemaNode(CamelModel):
    name: str
    tables: list[str] = []


class SchemaTree(CamelModel):
    database_name: str
    schemas: list[SchemaNode] = []

File: frontend/src/types/dataset.ts
export interface ColumnInfo {
  name: string
  type: string
  isDimension: boolean
  isMetric: boolean
  filterable: boolean
}

export interface DatasetInfo {
  id: string
  name: string
  tableName: string
  databaseId: string
  columns?: ColumnInfo[]
  rowCount?: number | null
}

export interface SchemaNode {
  name: string
  tables: string[]
}

export interface SchemaTree {
  databaseName: string
  schemas: SchemaNode[]
}
